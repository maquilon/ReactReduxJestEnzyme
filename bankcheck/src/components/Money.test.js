import React from 'react';
import { mount, shallow } from 'enzyme';
import { Money } from './Money';

describe('Money', () => {
    const mockFetchbitcoin = jest.fn();
    let props = { balance: 10, bitcoin: {} }
    let money = shallow(<Money {...props} />);

    it('renders properly', () => {
        expect(money).toMatchSnapshot();
    });

    describe('when mounted', () => {
        beforeEach(() => {
            props.fetchBitcoin = mockFetchbitcoin;
            money = mount(<Money {...props} />);
        })

        it('dispatches the `fetchBitcoin()` method it receives from props', () => {
            expect(mockFetchbitcoin).toHaveBeenCalled();
        });
    });

    describe('when there are valid bitcoin props', () => {
        beforeEach(() => {
            props = { balance: 10, bitcoin: { bpi: { USD: { rate: '1,000' } } } };
            money = shallow(<Money {...props} />)
        })

        it('displays the correct bitcoin value', () => {
            expect(money.find('h3').text()).toEqual('Bitcoin balance: 0.01');
        })
    });
});