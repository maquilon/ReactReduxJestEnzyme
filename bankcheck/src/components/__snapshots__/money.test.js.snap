// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Money renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Money
    balance={10}
    bitcoin={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Bitcoin balance: ",
        "",
      ],
      "style": Object {
        "marginTop": 50,
      },
    },
    "ref": null,
    "rendered": Array [
      "Bitcoin balance: ",
      "",
    ],
    "type": "h3",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Bitcoin balance: ",
          "",
        ],
        "style": Object {
          "marginTop": 50,
        },
      },
      "ref": null,
      "rendered": Array [
        "Bitcoin balance: ",
        "",
      ],
      "type": "h3",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
